### PR TITLE
[rllib] current learning rate update on local worker

### LINKED ISSUE #15910: [rllib] cur_lr not updated on local worker when LearningRateSchedule used on APEX DQN
### What is the problem?

`cur_lr` isn't updated on local worker when `lr_schedule` is set in configuration. Visible impact `cur_lr` in training stats is not updating.

*Ray version and other system information (Python version, TensorFlow version, OS):*
- Ray: 1.3.0
- Tensorflow: 2.5.0

### Reproduction (REQUIRED)

```python
import ray
from ray.rllib import agents

ray.init()
config = {
          'lr': 1e-4,
          'lr_schedule': [
                [0, 1e-4],
                [1000, 1e-9],
          ],
          'num_workers': 2,
          'train_batch_size': 10,
          'learning_starts': 10,
          'timesteps_per_iteration': 10,
}
trainer = agents.dqn.ApexTrainer(env='CartPole-v0', config=config)
while True:
    results = trainer.train()
    print(results['iterations_since_restore'],
          results['info']['learner']['default_policy']['cur_lr'])
```

Watch statistics printed:

```
1 9.999999747378752e-05
2 9.999999747378752e-05
3 9.999999747378752e-05
```

- [X] I have verified my script runs in a clean environment and reproduces the issue.
- [X] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

### PR DESCRIPTION
**Pull request by antoine-galataud**
Closes #15910. Tested with reproduction example given in issue and custom env.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [X] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by sven1977 — APPROVED** [review]
Cool, thanks a lot for the fix @antoine-galataud !

**Comment by sven1977** [comment]
@antoine-galataud 
Merged with master, just waiting for all tests to pass now.
